webapp/channels/src/components/markdown_image_expand/index.ts

```diff
@@ -1,28 +1,37 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {bindActionCreators, Dispatch} from 'redux';
+import {ReactNode} from 'react';
+import {connect, ConnectedProps} from 'react-redux';
 
-import {connect} from 'react-redux';
-
-import {GenericAction} from 'mattermost-redux/types/actions';
+import {Post} from '@mattermost/types/posts';
 
 import {toggleInlineImageVisibility} from 'actions/post_actions';
 import {isInlineImageVisible} from 'selectors/posts';
 import {GlobalState} from 'types/store';
 
-import MarkdownImageExpand, {Props} from './markdown_image_expand';
+import MarkdownImageExpand from './markdown_image_expand';
+
+export type OwnProps = {
+    postId: Post['id'];
+    imageKey: string;
+    alt: string;
+    onToggle?: (visible: boolean) => void;
+    children: ReactNode;
+}
 
-const mapStateToProps = (state: GlobalState, {postId, imageKey}: Props) => {
+const mapStateToProps = (state: GlobalState, {postId, imageKey}: OwnProps) => {
     return {
         isExpanded: isInlineImageVisible(state, postId, imageKey),
     };
 };
 
-const mapDispatchToProps = (dispatch: Dispatch<GenericAction>) => {
-    return {
-        actions: bindActionCreators({toggleInlineImageVisibility}, dispatch),
-    };
+const mapDispatchToProps = {
+    toggleInlineImageVisibility,
```

Comment:
Surprise, we don't need to [bindActionsCreator](https://redux.js.org/api/bindactioncreators#:~:text=The%20only%20use%20case%20for,dispatch%20wrapped%20function%20in%20return.)